docs/mp/lra.adoc

```diff
@@ -0,0 +1,460 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= LRA
+:toc:
+:toc-placement: preamble
+:description: Long Running Actions
+:keywords: helidon, mp, lra
+:h1Prefix: MP
+:feature-name: Long Running Actions
+:spec-version: 1.0-RC3
+:spec-name: Micro Profile {feature-name} specification
+:javadoc-link: https://download.eclipse.org/microprofile/microprofile-lra-{spec-version}/apidocs/org/eclipse/microprofile/lra/annotation/
+:rootdir: {docdir}/..
+
+== Contents
+
+- <<Overview, Overview>>
+- <<Maven Coordinates, Maven Coordinates>>
+- <<Usage, Usage>>
+- <<API, API>>
+- <<Configuration, Configuration>>
+- <<Examples, Examples>>
+- <<Additional Information, Additional Information>>
+- <<Reference, Reference>>
+
+== Overview
+
+Distributed transactions for microservices are known as SAGA design patterns and are defined by the {microprofile-lra-spec-url}[{spec-name}].
+Unlike well known XA protocol, LRA is asynchronous and therefore much more scalable. Every LRA JAX-RS resource (<<Participant, participant>>) defines endpoints to be invoked when transaction needs to be _completed_ or _compensated_.
+
+
+include::{rootdir}/includes/dependencies.adoc[]
+[source,xml]
+----
+<dependency>
+  <groupId>io.helidon.microprofile.lra</groupId>
+  <artifactId>helidon-microprofile-lra</artifactId>
+</dependency>
+<!-- Support for Narayana coordinator -->
+<dependency>
+  <groupId>io.helidon.lra</groupId>
+  <artifactId>helidon-lra-coordinator-narayana-client</artifactId>
+</dependency>
+----
+
+== Usage
+
+LRA transactions need to be coordinated over REST API by the LRA coordinator. <<Coordinator, Coordinator>>
+keeps track of all transactions and calls the `@Compensate` or `@Complete` endpoints for all participants involved in the particular
+transaction. LRA transaction is first started, then joined by <<Participant, participant>>.
+Participant reports the successful finish of transaction by calling complete. Coordinator then calls the JAX-RS
+_complete_ endpoint that was registered during the join of each
+<<Participant, participant>>. As the completed or compensated participants don't have to be on same instance,
+the whole architecture is highly scalable.
+
+image::lra/lra-complete-lb.svg[Complete]
+
+In case of error during the LRA transaction, participant reports cancel of LRA to coordinator.
+<<Coordinator, Coordinator>> calls compensate on all the joined participants.
+
+image::lra/lra-compensate-lb-error.svg[Cancel]
+
+When participant joins the LRA with timeout defined `@LRA(value = LRA.Type.REQUIRES_NEW, timeLimit = 5, timeUnit = ChronoUnit.MINUTES)`, coordinator compensate if timeout occurs before close is reported by participants.
```

Comment:
When a participant joins ...., the coordinator compensates if the timeout occurred before the close is reported by the participants.

Reply:
fixed)